yt/frontends/enzo/io.py

```diff
@@ -75,15 +78,21 @@ def _read_particle_fields(self, chunks, ptf, selector):
                     continue
                 ds = f.get("/Grid%08i" % g.id)
                 for ptype, field_list in sorted(ptf.items()):
-                    if ptype != "io":
+                    if ptype == "io":
+                        if g.NumberOfParticles == 0:
+                            continue
+                        pds = ds
+                    elif ptype == "DarkMatter":
                         if g.NumberOfActiveParticles[ptype] == 0:
                             continue
-                        pds = ds.get(f"Particles/{ptype}")
-                    else:
                         pds = ds
+                    else:
+                        if g.NumberOfActiveParticles[ptype] == 0:
+                            continue
+                        pds = ds.get("Active Particles/%s" % ptype)
                     pn = _particle_position_names.get(ptype, r"particle_position_%s")
                     x, y, z = (
-                        np.asarray(pds.get(pn % ax)[()], dtype="=f8") for ax in "xyz"
+                        np.asarray(pds.get(pn % ax).value, dtype="=f8") for ax in "xyz"
```

Comment:
The use of `.value` has been removed in newer versions of h5py. This could be what's causing the problem. If you switch it back to how it was, this should get things working.
```suggestion
                        np.asarray(pds.get(pn % ax)[()], dtype="=f8") for ax in "xyz"
```